From 1a50cfeedcf5faf741094faa3db1715ab89eaf9d Mon Sep 17 00:00:00 2001
Message-ID: <1a50cfeedcf5faf741094faa3db1715ab89eaf9d.1732384228.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1732384228.git.sharlatanus@gmail.com>
References: <cover.1732384228.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 21 Nov 2024 21:39:01 +0000
Subject: [PATCH 44/46] gnu: python-yt: Update to 4.4.0.

* gnu/packages/astronomy.scm (python-yt): Update to 4.4.0.
[arguments]<test-flags>: Disable some failing tests and tests requiring
network access.
[propagated-inputs]: Add python-astropy, python-cartopy,
python-glue-core, python-h5py, python-netcdf4, python-pandas,
python-pooch, python-regions, python-requests, and python-xarray.
[native-inputs]: Remove python-nose, python-nose-exclude, and
python-nose-timer; add nss-certs-for-test.

Change-Id: Ia868892b81fb8c5fc72ae20040b5f70ef936ff78
---
 gnu/packages/astronomy.scm | 42 ++++++++++++++++++++++++++++++--------
 1 file changed, 34 insertions(+), 8 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index aec9cc330a..858fb75d45 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -6584,19 +6584,30 @@ (define-public python-wiimatch
 (define-public python-yt
   (package
     (name "python-yt")
-    (version "4.3.1")
+    (version "4.4.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "yt" version))
        (sha256
-        (base32 "03jy35vyniyd1pd3sv0zpd2f3ks2iyqw65xv28ids8nw6v1vavbv"))))
+        (base32 "1mf084wp0wrhi17npzbyz42lh3xwlc2hggr8h4pmighsnafdy58f"))))
     (build-system pyproject-build-system)
     (arguments
      (list
       #:build-backend "setuptools.build_meta"
       #:test-flags
-      #~(list "--numprocesses" (number->string (parallel-job-count)))
+      #~(list "--numprocesses" (number->string (parallel-job-count))
+              "-k" (string-join
+                    ;; Tests requiring networking.
+                    (list "not test_load_sample_timeout"
+                          "test_unknown_filename"
+                          "test_typo_filename"
+                          "test_data_dir_broken"
+                          "test_registry_byte_size_dtype"
+                          "test_registry_byte_size_sign"
+                          ;; AssertionError: Arrays are not equal
+                          "test_field_cut_off_axis_octree")
+                    " and not "))
       #:phases
       #~(modify-phases %standard-phases
          (add-after 'unpack 'relax-requirements
@@ -6612,10 +6623,8 @@ (define-public python-yt
               (setenv "HOME" "/tmp")
               (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (native-inputs
-     (list python-cython-3
-           python-nose
-           python-nose-exclude
-           python-nose-timer
+     (list nss-certs-for-test
+           python-cython-3
            python-pyaml
            python-pytest
            python-pytest-mpl
@@ -6634,7 +6643,24 @@ (define-public python-yt
            python-pillow
            python-tomli-w
            python-tqdm
-           python-unyt))
+           python-unyt
+           ;; Optional
+           ;; python-arm-pyart ; not packed yet
+           ;; python-f90nml
+           ;; python-firefly
+           ;; python-miniball
+           ;; python-pyx
+           ;; python-ratarmount
+           python-astropy
+           python-cartopy
+           python-glue-core
+           python-h5py
+           python-netcdf4
+           python-pandas
+           python-pooch
+           python-regions
+           python-requests
+           python-xarray))
     (home-page "http://yt-project.org/")
     (synopsis "Analyzing and visualizing volumetric data framework")
     (description
-- 
2.46.0

